Server/scripts/fetchYearMaps.ts

```diff
@@ -88,15 +81,70 @@ const getBeatmapSet = memoizee(async (beatmap: APIBeatmap): Promise<Beatmapset>
         }
 
         // Check if new exists (remove if it does)
-        if (user.otherNames.some(v => v.name === beatmap.creator)) {
-            await user.otherNames.find(v => v.name === beatmap.creator)!.remove();
-            user.otherNames = user.otherNames.filter(v => v.name !== beatmap.creator);
+        if (user.otherNames.some(v => v.name === targetUser.username)) {
+            await user.otherNames.find(v => v.name === targetUser.username)!.remove();
+            user.otherNames = user.otherNames.filter(v => v.name !== targetUser.username);
         }
-        user.osu.username = beatmap.creator;
+        user.osu.username = targetUser.username;
         await user.save();
     }
+
+    return user;
+}, {
+    max: 200,
+    normalizer: ([user]) => `${user.username}`,
+});
+
+// Memoized method to create or fetch a BeatmapSet from DB.
+const existingSets: number[] = [];
+const getBeatmapSet = memoizee(async (beatmap: APIBeatmap): Promise<Beatmapset> => {
+    if(existingSets.includes(beatmap.setId))
```

Comment:
You might want to use a `Set` for `existingSets`, if the only purpose is checking for existence.
Something like 
```suggestion
const existingSets: Set<number> = new Set();
const getBeatmapSet = memoizee(async (beatmap: APIBeatmap): Promise<Beatmapset> => {
    if(existingSets.has(beatmap.setId))
```

`includes` gets a fair bit slower the bigger an array gets, but `has` is constant-time.

Note: you'll have to change `push` to `add` here: https://github.com/Corsace/Corsace/pull/133/files#diff-540ddd3d16c96190a6e5b286aec54346c93c98083adaf1644921ccedcb8cad7eR150